### ISSUE TITLE
Cache blocks in middleware and preapre paginated API

### ISSUE THREAD

**Issue by hoffmannjan** [casper-network/casper-blockexplorer#131]
So in this task, please prepare an API that will be used to query blocks from the frontend.

In the middleware:
- it should cache the responses from [SSE Sidecar](https://github.com/CasperLabs/event-sidecar/blob/dev/USAGE.md) or RPC and store the blocks history in node-cache.
- it should paginate the responses so when user will call the endpoint

```
GET /api/blocks?numPerPage=10
```

he will receive

```
{
  blocks: [...],
  numOfPages: 10929,
  numPerPage: 10,
  lastUpdated: timestamp
}
```

in the middleware it will be easy to calculcate the number of pages.
The stored blocks info in cache should be lazy - I mean it will be fetched only when the first user will ask for it. Then we will rely on cache (as its historical data we don't need to invalidate it).

Then please switch frontend to use this methods and refresh every 30s.

In future, its possible that SSE sidecar will have pagination functionality too so we will be able to remove this part. But for now please implement this so frontend won't have problems with rerendering etc.

**Comment by RitaMAllenCA** [comment]
@hoffmannjan would we want the explorer to ever go to the RPC? My understanding is the sidecar is there to avoid putting these burdens on the node - thoughts?

**Comment by hoffmannjan** [comment]
@RitaMAllenCA for now only in cases we can't get something through sidecar :). But now will move all request from frontend to middleware and point them to sidecar instead of RPC, indeed.

**Comment by RitaMAllenCA** [comment]
@hoffmannjan Hi - IIRC, the event store will be in the sidecar.  If so, can we close this as a "won't do" or do we still need this in the MW?

**Comment by hoffmannjan** [comment]
@RitaMAllenCA in fact it's a part of https://github.com/casper-network/casper-blockexplorer/pull/137 :)
